engine/lib/cache.php

```diff
@@ -234,4 +244,15 @@ function _elgg_cache_init() {
 
 return function(\Elgg\EventsService $events, \Elgg\HooksRegistrationService $hooks) {
 	$events->registerHandler('ready', 'system', '_elgg_cache_init');
+	
+	// register plugin hooks for cache reset
+	$events->registerHandler('cache:flush', 'system', 'elgg_reset_system_cache');
+	$events->registerHandler('cache:flush', 'system', 'elgg_invalidate_simplecache');
+	
+	// register some PHP caches for reset
+	// @see http://php.net/manual/en/refs.basic.php.php
+	$events->registerHandler('cache:flush', 'system', 'apc_clear_cache');
+	$events->registerHandler('cache:flush', 'system', 'opcache_reset');
+	$events->registerHandler('cache:flush', 'system', 'wincache_refresh_if_changed');
+	$events->registerHandler('cache:flush', 'system', 'wincache_ucache_clear');
```

Comment:
maybe add a code comment that during the event handling it is checked if the function is callable..